plugin/src/test/java/io/jenkins/plugins/casc/misc/ExportImportRoundTripAbstractTest.java

```diff
@@ -0,0 +1,260 @@
+     * 8.  Write the Jenkins configuration to $JENKINS_ROOT/jenkins.yaml.
+     * 9.  Restart Jenkins.
+     * 10. Check the {@link #stringInLogExpected()} is set during the restart.
+     *
+     * @throws IOException If an exception is thrown managing resources or files.
+     */
+    @Test
+    public void exportImportRoundTrip() throws IOException {
+        String resourcePath = configResource();
+        String resourceContent = getResourceContent(resourcePath);
+
+        assertNotNull(resourcePath);
+        assertNotNull(resourceContent);
+
+        r.then(step -> {
+            // Configure and validate
+            configureWithResource(resourcePath);
+            assertConfiguredAsExpected(r, resourceContent);
+
+            // Get the full configuration
+            //String jenkinsConf = getJenkinsConfViaWebUI();
+            //hack: the full Jenkins config fails due to defaultProperties of maven, we use the config of the plugin
+            //TODO: remove when https://issues.jenkins-ci.org/browse/JENKINS-57122 is solved
+            //@Issue("JENKINS-57122")
+            String jenkinsConf = getResourceContent(resourcePath);
+
+            // Get the schema
+            //String schema = getSchemaViaWebUI();
+
+            // Verify it's compliant to the schema
+            // TODO: when YAML schema validation is mature enough we can do that.
+            //verifyJsonAgainstSchema(jenkinsConf, schema);
+
+            // Check if the exported configuration is valid
+            assertConfigViaWebUI(jenkinsConf);
+
+            // Apply full configuration. Maybe not needed, we already have it configured and checked it is valid.
+            applyConfigViaWebUI(jenkinsConf);
+            assertConfiguredAsExpected(r, resourceContent);
+
+            // Configure Jenkins default JCasC file with the config file. It's already established, we check if applied
+            // looking at the logs.
+            putConfigInHome(jenkinsConf);
+
+            // Before restarting we need to establish this property, it's not managed properly so far
+            System.setProperty(ConfigurationAsCode.CASC_JENKINS_CONFIG_PROPERTY, new File(r.home, ConfigurationAsCode.DEFAULT_JENKINS_YAML_PATH).toURI().toURL().toExternalForm());
+
+            // Start recording the logs just before restarting, to avoid capture the previous startup. We're look there
+            // if the "magic token" is there
+            logging.record(DataBoundConfigurator.class.getName(), Level.INFO).capture(5);
+        });
+
+        r.then(step -> {
+            try {
+                // Verify the log shows it's configured
+                assertLogAsExpected(stringInLogExpected());
+
+                // Verify the configuration set at home/jenkins.yaml is loaded
+                assertConfiguredAsExpected(r, resourceContent);
+            } finally {
+                restoreJenkinsConfig();
```

Comment:
What if the failure happens during the first step?

Reply:
My understanding is that each execution should have its own Jenkins Rule running, so it should not impact. The test only would fail. Am I right @MRamonLeon ?

Reply:
You're right. But the `System.setProperty` may affect other tests. Anyway, I've removed the setter of the property after other fixes because it was failing from the beginning and it seems to work fine. So, after removing that I can safely remove the backup and restore code.